File: chdsr/chdsr-model/src/main/java/io/iochord/dev/chdsr/model/sbpnet/v1/Arc.java
package io.iochord.dev.chdsr.model.sbpnet.v1;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import io.iochord.dev.chdsr.model.sbpnet.v1.impl.ArcImpl;

@JsonDeserialize(as = ArcImpl.class)
public interface Arc extends Element {
	Element getSource();

	Element getTarget();
}

File: chdsr/chdsr-simulator/src/main/scala/io/iochord/dev/chdsr/simulator/model/Arc.scala
package io.iochord.dev.chdsr.simulator.model

object Direction extends Enumeration {
	  val In, Out = Value
}

case class Arc[T] (
  id: String,
  place: Place[T],
  transition: Transition,
  direction: Direction.Value)
  extends Element {
  
}